tests/autologging/fixtures.py

```diff
@@ -1,20 +1,33 @@
+import os
+
 import pytest
-from unittest import mock
 
-import mlflow.utils.autologging_utils as autologging_utils
+from mlflow.utils.autologging_utils import _is_testing, _AUTOLOGGING_TEST_MODE_ENV_VAR
 
 
 @pytest.fixture
 def test_mode_off():
-    with mock.patch("mlflow.utils.autologging_utils._is_testing") as testing_mock:
-        testing_mock.return_value = False
-        assert not autologging_utils._is_testing()
+    try:
+        prev_env_var_value = os.environ.pop(_AUTOLOGGING_TEST_MODE_ENV_VAR, None)
```

Comment:
I've refactored these fixtures to move away from mocking `autologging_utils._is_testing`, which is only effective if a module calls `autologging_utils._is_testing` without re-importing the `_is_testing` function. If we rely purely on environment variables (which are unit tested in the autologging safety suite), mocking challenges are avoided.